dotCMS/src/main/java/com/dotcms/rest/ContentResource.java

```diff
@@ -714,6 +718,21 @@ private Response multipartPUTandPOST(HttpServletRequest request, HttpServletResp
                     .equals("json")) {
                 try {
                     processJSON(contentlet, part.getEntityAs(InputStream.class));
+                    try {
+                        binaryFieldsInput =
+                            webResource.processJSON(part.getEntityAs(InputStream.class)).get("binary_fields")
+                                .toString();
+                    } catch (NullPointerException npe) {
+                    }
+                    if (UtilMethods.isSet(binaryFieldsInput)) {
+                        if (!binaryFieldsInput.contains(",")) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)